### PR TITLE
[MRV2][Spec Decode] Profile the rejection sampler to fix spec-decode OOM

### PR DESCRIPTION
**Pull request by mgoin**
## Purpose

The initial MRV2 memory profiling runs the plain `Sampler`, but spec decode uses the `RejectionSampler`, whose `apply_sampling_params` copies the expanded (`num_reqs * decode_query_len`) logits to a new fp32 tensor. That copy can be ~4GB for a standard Qwen3-8B DFlash model deployment.
  
This PR runs the rejection sampler in `profile_run` (via a spec-shaped `make_dummy`) so the peak is measured and KV cache is sized accordingly.

## Test Result

Using H100:

`vllm serve Qwen/Qwen3-8B --spec-model RedHatAI/Qwen3-8B-speculator.dflash --spec-method dflash --spec-tokens 7 --speculative-config '{"attention_backend":"FLASHINFER"}'` (using flashinfer tends to make the non-profiled space even tighter)

- Before: `Available KV cache: 52.02 GiB`, peak activation 3.04 GiB -> resulting in OOM in warmup
- After: `Available KV cache: 47.94 GiB`, peak activation 7.12 GiB -> serves normally

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [ ] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [ ] The test plan, such as providing test command.
- [ ] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
</details>

### CODE REVIEW

File: vllm/v1/worker/gpu/model_runner.py [hunk 1]
```diff
@@ -639,6 +643,24 @@ def _dummy_sampler_run(self, hidden_states: torch.Tensor) -> None:
...
+    @torch.inference_mode()
+    def _dummy_rejection_sampler_run(self, hidden_states: torch.Tensor) -> None:
+        assert self.rejection_sampler is not None
+        assert self.speculator is not None
+        # The rejection sampler copies the expanded decode_query_len logits per request
+        num_logits_per_req = self.decode_query_len
+        num_reqs = max(
+            1, min(self.max_num_reqs, self.max_num_tokens // num_logits_per_req)
+        )
+        num_logits = num_reqs * num_logits_per_req
```
Comment (mgoin) [inline vllm/v1/worker/gpu/model_runner.py:656]: This may be too conservative, not sure the "right" approach here

**Comment by mgoin** [comment]
Let's go with https://github.com/vllm-project/vllm/pull/48641
